File: Examples/test-suite/python/typedef_funcptr_runme.py
import typedef_funcptr

a = 100
b = 10

if typedef_funcptr.do_op(a,b,typedef_funcptr.addf) != 110:
  raise RuntimeError("addf failed")
if typedef_funcptr.do_op(a,b,typedef_funcptr.subf) != 90:
  raise RuntimeError("subf failed")

if typedef_funcptr.do_op_typedef_int(a,b,typedef_funcptr.addf) != 110:
  raise RuntimeError("addf failed")
if typedef_funcptr.do_op_typedef_int(a,b,typedef_funcptr.subf) != 90:
  raise RuntimeError("subf failed")

if typedef_funcptr.do_op_typedef_Integer(a,b,typedef_funcptr.addf) != 110:
  raise RuntimeError("addf failed")
if typedef_funcptr.do_op_typedef_Integer(a,b,typedef_funcptr.subf) != 90:
  raise RuntimeError("subf failed")

File: Examples/test-suite/javascript/typedef_funcptr_runme.js
var typedef_funcptr = require("typedef_funcptr");

var a = 100;
var b = 10;

if (/* await */(typedef_funcptr.do_op(a, b, typedef_funcptr.addf)) != 110) {
  throw new Error("addf failed");
}
if (/* await */(typedef_funcptr.do_op(a, b, typedef_funcptr.subf)) != 90) {
  throw new Error("subf failed");
}

if (/* await */(typedef_funcptr.do_op_typedef_int(a, b, typedef_funcptr.addf)) != 110) {
  throw new Error("addf failed");
}
if (/* await */(typedef_funcptr.do_op_typedef_int(a, b, typedef_funcptr.subf)) != 90) {
  throw new Error("subf failed");
}

if (/* await */(typedef_funcptr.do_op_typedef_Integer(a, b, typedef_funcptr.addf)) != 110) {
  throw new Error("addf failed");
}
if (/* await */(typedef_funcptr.do_op_typedef_Integer(a, b, typedef_funcptr.subf)) != 90) {
  throw new Error("subf failed");
}
